src/drivers/cloudflare-kv-http.ts

```diff
@@ -97,44 +102,46 @@ export default defineDriver<KVHTTPOptions>((opts) => {
 
   const apiURL = opts.apiURL || "https://api.cloudflare.com";
   const baseURL = `${apiURL}/client/v4/accounts/${opts.accountId}/storage/kv/namespaces/${opts.namespaceId}`;
-  const kvFetch = $fetch.create({ baseURL, headers });
+
+  let kvFetch = async (url: string, fetchOptions?: FetchOptions) =>
+    $fetch.native(`${baseURL}${url}`, {
+      headers,
+      ...(fetchOptions as any),
+    });
+
+  //@ts-expect-error fetch needs to be used directly for msw to work
+  if (opts.apiToken === "msw") {
+    kvFetch = async (url: string, fetchOptions?: FetchOptions) =>
+      fetch(`${baseURL}${url}`, { headers, ...(fetchOptions as any) });
+  }
 
   const hasItem = async (key: string) => {
-    try {
-      const res = await kvFetch(`/metadata/${key}`);
-      return res?.success === true;
-    } catch (err: any) {
-      if (!err?.response) {
-        throw err;
-      }
-      if (err?.response?.status === 404) {
-        return false;
-      }
-      throw err;
-    }
+    const response = await kvFetch(`/metadata/${key}`);
```

Comment:
This is a `test:` PR. I would appreciate if you can include any possible refactors you need in a separate PR with clear description why we need any of those.

Reply:
Apologies if this looks weird as this includes #255 changes too. The description of why it's needed is in the PR description.